src/ucp/api/device/ucp_device_impl.h

```diff
@@ -93,8 +93,8 @@ UCS_F_DEVICE ucs_status_t ucp_device_prepare_send(
 
     device_ep   = mem_list_h->uct_device_eps[lane];
     elem_offset = first_mem_elem_index * mem_list_h->uct_mem_element_size[lane];
-    uct_elem    = (uct_device_mem_element_t*)UCS_PTR_BYTE_OFFSET(mem_list_h + 1,
-                                                                 elem_offset);
+    uct_elem    = (uct_device_mem_element_t*)
+            UCS_PTR_BYTE_OFFSET(mem_list_h->uct_mem_elements, elem_offset);
```

Comment:
maybe save uct_mem_elements per lane, and remove the field uct_mem_element_size ?